### PR TITLE
feat(defaults): visual CTRL-R for LSP mappings

### LINKED ISSUE #28528: New lsp keybindings: `crr` in visual mode breaks visual selection -> `c` workflow
### Problem

The new lsp keybindings include `crr` in visual mode for code actions. This breaks the visual mode -> `c` workflow because the user now has to wait for `timeoutlen`.

### Steps to reproduce

After a server has attached, select some text, then press `c`

### Expected behavior

Selected text is gone and nvim is in insert mode immediately

### Neovim version (nvim -v)

0.10.0 commit 9b8a0755390b7eb3ad369f3a0a42eb9aecb8cbe2

### Vim (not Nvim) behaves the same?

no

### Operating system/version

N/A

### Terminal name/version

N/A

### $TERM environment variable

N/A

### Installation

N/A

### PR DESCRIPTION
**Pull request by justinmk**
fix #28528

### CODE REVIEW

File: runtime/lua/vim/_defaults.lua [hunk 1]
```diff
@@ -159,7 +159,16 @@ do
       vim.lsp.buf.rename()
     end, { desc = 'vim.lsp.buf.rename()' })
 
-    vim.keymap.set({ 'n', 'v' }, 'crr', function()
```
Comment (justinmk) [inline runtime/lua/vim/_defaults.lua:162]: was 'v' instead of 'x' intentional?

Comment (gpanders) [reply runtime/lua/vim/_defaults.lua:162]: No

**Review by mfussenegger — APPROVED** [review]
Personally I'd prefer a symmetric mapping, but the rationale behind this makes sense, and I don't have a better sugestion either, so :+1:

File: runtime/lua/vim/_defaults.lua (hunk 1 above)
Comment (echasnovski) [reply runtime/lua/vim/_defaults.lua:162]: Besides, `<C-r>` in Select mode is a thing, so using `'x'` mode is probably safer.

**Review by gpanders — APPROVED** [review]

File: runtime/doc/lsp.txt [hunk 2]
```diff
@@ -61,10 +61,12 @@ options are not restored when the LSP client is stopped or detached.
 - |K| is mapped to |vim.lsp.buf.hover()| unless |'keywordprg'| is customized or
   a custom keymap for `K` exists.
 
-                                                  *crr* *v_crr* *crn* *i_CTRL-S*
+                 *crr* *v_crr* *crn* *i_CTRL-S* *v_CTRL-R_CTRL-R* *v_CTRL-R_r*
```
Comment (kang8) [inline runtime/doc/lsp.txt:64]: It seems that **v_crr** needs to be removed?

```suggestion
                 *crr* *crn* *i_CTRL-S* *v_CTRL-R_CTRL-R* *v_CTRL-R_r*
```

**Pull request merged**